### PR TITLE
Remove lock in FunctionManager

### PR DESCRIPTION
**Pull request by xuguruogu**
atomic operation acts like dead lock, and can not go out of the while loop if multi-thread access it frequently.

image

### CODE REVIEW

**Comment by dutor** [comment]
But given your post of `perf top`, I am curious about why `FunctionManager::instance` is so hot. Except for the first several invokes, it does not involve any atomic or memory barrier thing.

**Comment by xuguruogu** [comment]
> But given your post of `perf top`, I am curious about why `FunctionManager::instance` is so hot. Except for the first several invokes, it does not involve any atomic or memory barrier thing.

call function as filter with a little part of data returned.

File: src/common/filter/FunctionManager.h [hunk 1]
```diff
@@ -60,7 +60,6 @@ class FunctionManager final {
         Function                body_;
     };
 
-    mutable folly::RWSpinLock                           lock_;
     std::unordered_map<std::string, FunctionAttributes> functions_;
```
Comment (Shylock-Hg) [inline src/common/filter/FunctionManager.h:63]: I think you better mark it const.

Comment (xuguruogu) [reply src/common/filter/FunctionManager.h:63]: This may not help due to ref counter use atomic operation.

**Pull request merged**